arangod/Pregel/WorkerConductorMessages.h

```diff
@@ -279,4 +256,78 @@ auto inspect(Inspector& f, CollectPregelResults& x) {
       f.field(Utils::executionNumberKey, x.executionNumber),
       f.field("withId", x.withId).fallback(false));
 }
+
+// ---------------------- modern message ----------------------
+
+using MessagePayload =
+    std::variant<LoadGraph, PrepareGlobalSuperStep, RunGlobalSuperStep>;
+using ResponsePayload =
+    std::variant<ResultT<GraphLoaded>, ResultT<GlobalSuperStepPrepared>,
+                 ResultT<GlobalSuperStepFinished>>;
+
+struct MessagePayloadSerializer : MessagePayload {};
+template<class Inspector>
+auto inspect(Inspector& f, MessagePayloadSerializer& x) {
+  return f.variant(x).unqualified().alternatives(
+      arangodb::inspection::type<LoadGraph>("loadGraph"),
+      arangodb::inspection::type<PrepareGlobalSuperStep>(
+          "prepareGlobalSuperStep"),
+      arangodb::inspection::type<RunGlobalSuperStep>("runGlobalSuperStep"));
+}
+struct ResponsePayloadSerializer : ResponsePayload {};
+template<class Inspector>
+auto inspect(Inspector& f, ResponsePayloadSerializer& x) {
+  return f.variant(x).unqualified().alternatives(
+      arangodb::inspection::type<ResultT<GraphLoaded>>("graphLoaded"),
+      arangodb::inspection::type<ResultT<GlobalSuperStepPrepared>>(
+          "globalSuperStepPrepared"),
+      arangodb::inspection::type<ResultT<GlobalSuperStepFinished>>(
+          "globalSuperStepFinished"));
+}
+template<typename Inspector>
+auto inspect(Inspector& f, MessagePayload& x) {
+  if constexpr (Inspector::isLoading) {
+    auto v = MessagePayloadSerializer{};
+    auto res = f.apply(v);
+    x = MessagePayload{v};
+    return res;
+  } else {
+    auto v = MessagePayloadSerializer{x};
+    return f.apply(v);
+  }
+}
+template<typename Inspector>
+auto inspect(Inspector& f, ResponsePayload& x) {
+  if constexpr (Inspector::isLoading) {
+    auto v = ResponsePayloadSerializer{};
+    auto res = f.apply(v);
+    x = ResponsePayload{v};
+    return res;
+  } else {
+    auto v = ResponsePayloadSerializer{x};
+    return f.apply(v);
+  }
+}
+
+struct ModernMessage {
```

Comment:
Right now `ModernMessage` and `ModernResponse` are the same struct; 

Can you get away with just having a `ModernMessage`? Or in other words: Is the fact that something is a Response *used*?

Reply:
Further: I think at the moment you encode in the Payload already that something is a response; maybe this messaging layer should not assume a request-response model, but rather a "get my messages around" model?

Reply:
Yes, it *is* relevant because the body in std::visit that is called in Worker::process has to be exhaustive. Here you just expect ModernMessages, no ModernResponses. (And it will be the other way around for the messages in the conductor as soon as I use the new messages in Feature::handleConductorRequest as well. 
Ok, you could pattern match only the relevant cases and do an else case for the rest, but isn't it more readable to separate the two? This way we always get a compile error if we forget to add a case when we added a new message to ModernMessage.

Reply:
We could also call it differently: E.g. MessageFromWorker and MessageFromConductor

Reply:
I am going to double down on the "just get my messages around already" model ;) 

If you go with just `ModernMessage`, then not only do you not duplicate code here, the message handlers become more uniform. Ultimately I would like the the concept of `handleConductorRequest` to not exist anymore, I would like the Conductor to handle messages sent to it.

At some point you will have to write code that says "I did not expect this message", and in my opinion it should be the conductor's responsibility to do so; maybe we can provide some small transformation helper that specifies which messages a conductor wants to handle, and do the erroring outside transparently.

I would like it to be as simple as possible to add new message types (and also messaging paradigms). It looks to me like this split makes it harder to do so, or even impossible if we are boxed into "request-response"